projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -1044,20 +1045,25 @@ export class IgxComboComponent implements AfterViewInit, ControlValueAccessor, O
      * @hidden
      */
     public isItemSelected(item) {
-        return this.selectionAPI.is_item_selected(this.id, item, this._compareFunc.apply(this, [item, true]));
+        return this.selectionAPI.is_item_selected(this.id, this._stringifyItemID(item));
     }
 
     /**
      * @hidden
      */
-    protected triggerSelectionChange(newSelection) {
+    protected triggerSelectionChange(newSelectionAsSet: Set<any>) {
         const oldSelection = this.dropdown.selectedItem;
+        const newSelection = newSelectionAsSet ? Array.from(newSelectionAsSet) : [];
         if (oldSelection !== newSelection) {
             const args: IComboSelectionChangeEventArgs = { oldSelection, newSelection };
             this.onSelectionChange.emit(args);
-            this.selectionAPI.set_selection(this.id, newSelection);
+            newSelectionAsSet = new Set();
+            for (let i = 0; i < args.newSelection.length; i++) {
```

Comment:
`newSelection` already is an array, I think it would be easier to read if cycling through `newSelection` instead of `args.newSelection`